integration-cli/docker_cli_service_logs_experimental_test.go

```diff
@@ -48,6 +49,100 @@ func (s *DockerSwarmSuite) TestServiceLogs(c *check.C) {
 	}
 }
 
+func (s *DockerSwarmSuite) TestServiceLogsCompleteness(c *check.C) {
+	testRequires(c, ExperimentalDaemon)
+	d := s.AddDaemon(c, true, true)
+
+	name := "TestServiceLogsCompleteness"
+
+	// make a service that prints 6 lines
+	out, err := d.Cmd("service", "create", "--name", name, "busybox", "sh", "-c", "for line in 1 2 3 4 5 6; do echo log test $line; done; sleep 100000")
+	c.Assert(err, checker.IsNil)
+	c.Assert(strings.TrimSpace(out), checker.Not(checker.Equals), "")
+
+	// make sure task has been deployed.
+	waitAndAssert(c, defaultReconciliationTimeout, d.CheckActiveContainerCount, checker.Equals, 1)
```

Comment:
you're almost certainly correct, and the possibility of a race also exists in the since test as well, as I've just realized (sleeping does not guarantee that anything else has happened on the system in the meantime.)